### PR TITLE
[Azure Search] Update relevancy tests

### PR DESCRIPTION
**Pull request by loic-sharma**
I've split tests into two categories:

1. "First results" - These tests that the expected results are first
2. "Top results" - These tests that the expected results are found

I ensured that our tests align with the most frequently selected results, as per our search telemetry.

Addresses https://github.com/nuget/nugetgallery/issues/6980

### CODE REVIEW

**Comment by joelverhagen** [comment]
Great idea on these two buckets 🥇

File: tests/NuGet.Services.AzureSearch.FunctionalTests/Relevancy/V3RelevancyFunctionalTests.cs [hunk 1]
```diff
@@ -14,65 +16,74 @@ public V3RelevancyFunctionalTests(CommonFixture fixture, ITestOutputHelper testO
...
         }
 
-        [RelevancyFact]
-        public async Task MicrosoftExtensions()
+        [RelevancyTheory]
+        [MemberData(nameof(EnsureTopResultsData))]
+        public async Task EnsureTopResults(string searchTerm, string[] expectedTopResults)
         {
-            var results = await SearchAsync("Microsoft.Extensions");
+            var results = await SearchAsync(searchTerm);
```
Comment (joelverhagen) [inline tests/NuGet.Services.AzureSearch.FunctionalTests/Relevancy/V3RelevancyFunctionalTests.cs]: i think the number of results returned should be clarified here. For example if 100 is returned, this is not a very strong test.

Comment (joelverhagen) [reply tests/NuGet.Services.AzureSearch.FunctionalTests/Relevancy/V3RelevancyFunctionalTests.cs]: Would prefer a parameter to `SearchAsync` but a comment can work too (but may be out of date). Perhaps an assertion too like asserting the count is less than or equal to something.

File: tests/NuGet.Services.AzureSearch.FunctionalTests/Relevancy/V3RelevancyFunctionalTests.cs [hunk 2]
```diff
@@ -14,65 +16,74 @@ public V3RelevancyFunctionalTests(CommonFixture fixture, ITestOutputHelper testO
...
+            // TODO: Reduce exact match boost and update "json.net", "entity framework", "redis", and "csv" tests.
+            // See: https://github.com/NuGet/NuGetGallery/issues/7330
+            yield return new object[] { "newtonsoft.json", new[] { "newtonsoft.json" } };
+            yield return new object[] { "newtonsoft", new[] { "newtonsoft.json" } };
+            yield return new object[] { "json.net", new[] { "json.net", "newtonsoft.json" } };
+            yield return new object[] { "json", new[] { "json", "newtonsoft.json" } };
 
-            Assert.NotEmpty(results);
-            Assert.Contains("log4net", results);
+            yield return new object[] { "tags:\"aws-sdk-v3\"", new[] { "awssdk.core", "awssdk.s3" } };
```
Comment (shishirx34) [inline tests/NuGet.Services.AzureSearch.FunctionalTests/Relevancy/V3RelevancyFunctionalTests.cs]: did you run this test against dev? I do not see a package "awssdk.s3" in search(at least in gallery)

Comment (shishirx34) [reply tests/NuGet.Services.AzureSearch.FunctionalTests/Relevancy/V3RelevancyFunctionalTests.cs]: oh are these tests only meant for prod? forgot.

File: tests/NuGet.Services.AzureSearch.FunctionalTests/Relevancy/V3RelevancyFunctionalTests.cs (hunk 1 above)
Comment (loic-sharma) [reply tests/NuGet.Services.AzureSearch.FunctionalTests/Relevancy/V3RelevancyFunctionalTests.cs]: Added a parameter and explicitly chose the number of results in the test

**Pull request merged**